From 3185f1f2cbe74eaddd80698619eba7bd87f69c36 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Wed, 7 Jan 2015 10:56:54 -0500
Subject: [redhat] kabi: add RH_KABI_DEPRECATE

Message-id: <bc3eab0b470cf2c47aaeb6748f0a1b3e58b9d45b.1420627472.git.jbenc@redhat.com>
Patchwork-id: 102510
O-Subject: [RHEL7.1 net PATCH 1/5] kabi: add RH_KABI_DEPRECATE
Bugzilla: 1173444
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Kyle McMartin <kmcmarti@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173444
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8460333
Upstream: RHEL only

Add a RH_KABI_DEPRECATE macro to mark a given element as deprecated.

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/rh_kabi.h b/include/linux/rh_kabi.h
index 8e964a6..4b00ee7 100644
--- a/include/linux/rh_kabi.h
+++ b/include/linux/rh_kabi.h
@@ -28,6 +28,8 @@
  * RH_KABI_REPLACE_P - replacement of _orig pointer with _new pointer.  Pointers
  *                     don't work with anonymous unions and their sizes don't
  *                     change, so just do a straightforward replacement.
+ * RH_KABI_DEPRECATE - mark the element as deprecated and make it unusable
+ *         by modules while preserving kABI checksums
  *
  * RH_KABI_EXTEND - simple macro for adding a new element to a struct while
  *                  preserving the kabi agreement (by wrapping with GENKSYMS).
@@ -47,6 +49,7 @@
 # define _RH_KABI_REPLACE(_orig, _new)  _orig
 # define _RH_KABI_CHANGE_TYPE(_orig, _new) _orig
 # define _RH_KABI_REPLACE_P(_orig, _new) _orig
+# define _RH_KABI_DEPRECATE(_type, _orig) _type _orig
 
 # define RH_KABI_EXTEND(_new)
 # define RH_KABI_FILL_HOLE(_new)
@@ -65,6 +68,8 @@
  }
 # define _RH_KABI_REPLACE_P(_orig, _new) _new
 
+# define _RH_KABI_DEPRECATE(_type, _orig) _type rh_reserved_##_orig
+
 # define RH_KABI_EXTEND(_new)           _new;
 
 /* Warning, only use if a hole exists for _all_ arches. Use pahole to verify */
@@ -76,6 +81,7 @@
 #define RH_KABI_REPLACE(_orig, _new)  _RH_KABI_REPLACE(_orig, _new);
 #define RH_KABI_CHANGE_TYPE(_orig, _new) _RH_KABI_CHANGE_TYPE(_orig, _new);
 #define RH_KABI_REPLACE_P(_orig, _new)  _RH_KABI_REPLACE_P(_orig, _new);
+#define RH_KABI_DEPRECATE(_type, _orig)  _RH_KABI_DEPRECATE(_type, _orig);
 
 /*
  * We tried to standardize on Red Hat reserved names.  These wrappers leverage
-- 
1.7.1